app/Services/AuthService.php

```diff
@@ -27,71 +31,206 @@ public function __construct(User $userObj)
+            ];
             return $data;
         }
 
-        $userOtp = $this->userOtpObj->where('user_id', $user['id'])->where('otp', $inputs['otp'])->where('otp_for', 'forget_password')->first();
+        $userOtp = $this->userOtpObj->whereUserId($user['id'])->whereOtp($inputs['otp'])->where('otp_for', 'reset_password')->first();
         if ($userOtp == null) {
-            $data['errors']['otp'][] = __('message.invalidOtp');
+            $data = [
+                'status' => false,
+                'message' => __('message.invalidOtp')
+            ];
 
             return $data;
         }
+
         $expirationTime = config('site.otpExpirationTimeInMinutes');
         $expirationDate = Carbon::parse($userOtp['created_at'])->addMinutes($expirationTime)->format('Y-m-d H:i:s');
 
-        if ($userOtp['used_at'] != null || date('Y-m-d h:i:s') > $expirationDate) {
-            $data['errors']['otp'][] = __('message.invalidOtp');
+        if ($userOtp['verified_at'] != null || date('Y-m-d h:i:s') > $expirationDate) {
+            $data = [
+                'status' => false,
+                'message' => __('message.invalidOtp')
+            ];
 
             return $data;
         }
-        $this->userOtpService->update($userOtp['id'], ['used_at' => date('Y-m-d h:i:s')]);
+
+        $this->userOtpService->update($userOtp['id'], ['verified_at' => date('Y-m-d h:i:s')]);
         $user->password = $inputs['password'];
         $user->save();
-        $data['message'] = __('message.passwordChangeSuccess');
 
+        $data = [
+            'status' => true,
+            'message' => __('message.passwordChangeSuccess')
+        ];
+
+        return $data;
+    }
+
+    public function generateOtp($inputs)
+    {
+        $user = $this->userObj->whereEmail($inputs['email'])->first();
+        if ($user == null) {
+            $data = [
+                'status' => false,
+                'message' =>  __('message.emailNotExist')
+            ];
+            return $data;
+        }
+
+        $otp = mt_rand(100000, 999999);
+
+        switch ($inputs['otp_for']) {
+            case 'verification':
+                $usedFor = 'SignUp';
+                break;
+            case 'update_profile':
+                $usedFor = 'Update Profile';
+                break;
+            case 'reset_password':
+                $usedFor = 'Forgot Password';
+                break;
+            default:
+                $usedFor = '';
+                break;
+        }
+
+        $this->userOtpService->store(['otp' => $otp, 'user_id' => $user['id'], 'otp_for' => $inputs['otp_for']]);
+        Mail::to($inputs['email'])->send(new SignUp(['otp' => $otp, 'used_for' => $usedFor]));
+        $data['otp'] = $otp;
+
+        $data = [
+            'status' => true,
+            'message' => 'Otp Send Successfully'
+        ];
+
+        return $data;
+    }
+
+    public function changePassword($inputs)
+    {
```

Comment:
@damini-7span Pls add condition to check current password are matched so login user password must changed.